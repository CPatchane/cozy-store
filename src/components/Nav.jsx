import React from 'react'
import { translate } from 'cozy-ui/react/I18n'
import { NavLink } from 'react-router-dom'

export const Nav = ({ t }) => (
  <nav>
    <ul className='coz-nav'>
      <li className='coz-nav-item'>
        <NavLink to='/discover' className='sto-cat-discover coz-nav-link' activeClassName='active'>
          { t('nav.discover') }
        </NavLink>
      </li>
      <li className='coz-nav-item'>
        <NavLink to='/myapps' className='sto-cat-myapps coz-nav-link' activeClassName='active'>
          { t('nav.myapps') }
        </NavLink>
      </li>
    </ul>
  </nav>
)

export default translate()(Nav)
