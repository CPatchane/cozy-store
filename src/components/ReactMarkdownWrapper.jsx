import React from 'react'
import ReactMarkdown from 'react-markdown'

export const reactMarkdownRendererOptions = {
  Link: props => <a href={props.href} target='_blank'>{props.children}</a>
}

export const ReactMarkdownWrapper = ({ source }) =>
  <ReactMarkdown
    source={source}
    renderers={reactMarkdownRendererOptions}
  />

export default ReactMarkdownWrapper
