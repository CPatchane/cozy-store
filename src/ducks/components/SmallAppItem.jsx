import React from 'react'

import Icon from 'cozy-ui/react/Icon'
import { translate } from 'cozy-ui/react/I18n'

import defaultAppIcon from '../../assets/icons/icon-cube.svg'
import { Link } from 'react-router-dom'
import Button from 'cozy-ui/react/Button'

export const SmallAppItem = ({
  t,
  slug,
  developer,
  editor,
  icon,
  name,
  namePrefix,
  installed,
  onClick,
  installedAppLink
}) => {
  const onShortcutAppButton = (e, link) => {
    // prevent from opening parent SmallAppItem link
    // which will lead to the application main page
    e.stopPropagation()
    if (link) window.location.assign(link)
  }
  return (
    // HACK a11y
    // `onKeyDown={(e) => e.keyCode === 13 ? onClick() : null`
    // This syntax make the div accessible.
    <div
      className='sto-small-app-item'
      onKeyDown={e => (e.keyCode === 13 ? onClick() : null)}
      onClick={onClick}
      tabIndex={0}
    >
      {icon ? (
        <img
          src={icon}
          alt={`${slug}-icon`}
          width='64'
          height='64'
          className='sto-small-app-item-icon'
        />
      ) : (
        <Icon
          className='sto-small-app-item-icon--default blurry'
          width='48'
          height='64'
          icon={defaultAppIcon}
          color='#95999D'
        />
      )}
      {installed && (
        <Icon
          className='sto-small-app-item-badge-installed'
          icon='check'
          color='#2bba40'
        />
      )}
      <div className='sto-small-app-item-desc'>
        <h4 className='sto-small-app-item-title'>
          {namePrefix && `${namePrefix} `}{name}
        </h4>
        <p className='sto-small-app-item-detail'>
          {developer.name === 'Cozy' ? 'Cozy Cloud Inc.' : developer.name}
        </p>
        {installed
          ? <Button
            onClick={(e) => onShortcutAppButton(e, installedAppLink)}
            label={t('app_item.open')}
            theme='secondary'
            className='sto-small-app-item-button'
            size='tiny'
          />
          : <Link
            to={`/discover/${slug}/manage`}
            onClick={(e) => onShortcutAppButton(e)}
            className='c-btn c-btn--regular c-btn--tiny sto-small-app-item-button sto-small-app-item-button-install'
          >
            <span>{t('app_item.install')}</span>
          </Link>
        }
      </div>
    </div>
  )
}

export default translate()(SmallAppItem)
