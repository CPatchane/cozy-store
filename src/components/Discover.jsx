import React, { Component } from 'react'

import { translate } from 'cozy-ui/react/I18n'

import SmallAppItem from './SmallAppItem'

const DEMO_APPS = [
  {
    slug: 'bank',
    name: 'Bank',
    developer: {name: 'Cozy'},
    editor: 'Cozy',
    icon: require('../assets/temp_apps/icon-bank.svg')
  },
  {
    slug: 'sante',
    name: 'Sant√©',
    developer: {name: 'MAIF Ltd'},
    editor: '',
    icon: require('../assets/temp_apps/icon-sante.svg')
  },
  {
    slug: 'mon_logis',
    name: 'Mon Logis',
    developer: {name: 'HoodBrains'},
    editor: '',
    icon: require('../assets/temp_apps/icon-mon_logis.svg')
  },
  {
    slug: 'calendar',
    name: 'Calendar',
    developer: {name: 'Cozy'},
    editor: 'Cozy',
    icon: require('../assets/temp_apps/icon-calendar.svg')
  }
]

class Discover extends Component {
  render () {
    const { t } = this.props
    return (
      <div className='sto-discover sto-soon'>
        <h2 className='sto-discover-title'>{t('discover.title')}</h2>
        <div className='sto-discover-get-started'>
          <h3 className='sto-discover-get-started-title'>
            {t('discover.get_started')}
          </h3>
          <div className='sto-discover-get-started-list'>
            {DEMO_APPS.map(a => {
              return <SmallAppItem
                slug={a.slug}
                developer={a.developer}
                editor={a.editor}
                icon={a.icon}
                name={a.name}
                version={a.version}
              />
            })}
          </div>
        </div>
      </div>
    )
  }
}

export default translate()(Discover)
