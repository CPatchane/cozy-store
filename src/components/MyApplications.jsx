import React, { Component } from 'react'

import { Route } from 'react-router-dom'
import { translate } from 'cozy-ui/react/I18n'
import Spinner from 'cozy-ui/react/Spinner'

import SmallAppItem from './SmallAppItem'
import ApplicationModal from './ApplicationModal'

class MyApplications extends Component {
  constructor (props) {
    super(props)
    props.fetchApps()
  }

  onAppClick (appSlug) {
    this.props.history.push(`/myapps/${appSlug}/manage`)
  }

  render () {
    const { t, myApps, isFetching, error } = this.props
    return (
      <div className='sto-myapps'>
        <h2 className='sto-myapps-title'>{t('myapps.title')}</h2>
        <div className='sto-myapps-list'>
          {!isFetching && myApps && !!myApps.length &&
            myApps.map(a => {
              return <SmallAppItem
                slug={a.slug}
                developer={a.developer}
                editor={a.editor}
                icon={a.icon}
                name={a.name}
                version={a.version}
                onClick={() => this.onAppClick(a.slug)}
              />
            })
          }
          {error &&
            <p>{error}</p>
          }
        </div>
        {isFetching &&
          <Spinner
            size='xxlarge'
            loadingType='appsFetching'
            middle='true'
          />
        }

        <Route path='/myapps/:appSlug/manage' render={({ match }) =>
          (<ApplicationModal {...this.props} match={match} />)
        } />
      </div>
    )
  }
}

export default translate()(MyApplications)
