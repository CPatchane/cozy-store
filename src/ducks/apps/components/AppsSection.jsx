import React from 'react'
import { translate } from 'cozy-ui/react/I18n'

import SmallAppItem from '../../components/SmallAppItem'
import { getLocalizedAppProperty } from '../index'

const _renderAppComponent = (app, lang, onAppClick) => {
  return (
    <SmallAppItem
      slug={app.slug}
      developer={app.developer || {}}
      namePrefix={getLocalizedAppProperty(app, 'name_prefix', lang) || ''}
      editor={app.editor || ''}
      icon={app.icon}
      name={getLocalizedAppProperty(app, 'name', lang)}
      installed={app.installed}
      onClick={() => onAppClick(app.slug)}
      installedAppLink={app.related}
      key={app.slug}
    />
  )
}

export const AppsSection = ({lang, appsList, subtitle, onAppClick}) => {
  return (
    <div className='sto-sections-apps'>
      {subtitle && <h3 className='sto-sections-subtitle'>
        {subtitle}
      </h3>}
      {appsList && !!appsList.length && <div className='sto-sections-list'>
        {appsList.map(
          app => _renderAppComponent(app, lang, onAppClick)
        )}
      </div>}
    </div>
  )
}

export default translate()(AppsSection)
