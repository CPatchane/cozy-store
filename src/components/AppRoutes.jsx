import React from 'react'
import { Route, Switch, Redirect } from 'react-router-dom'

import MyApplications from '../containers/MyApplications'

const AppRoutes = () => {
  return <Switch>
    <Route path='/myapps' component={MyApplications} />
    <Redirect from='/' to='/myapps' />
  </Switch>
}

export default AppRoutes
