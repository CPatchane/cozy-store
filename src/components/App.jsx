
import React from 'react'

import Sidebar from './Sidebar'
import AppRoutes from './AppRoutes'

const App = () => (
  <div className='sto-wrapper coz-sticky'>
    <Sidebar />
    <main className='sto-content'>
      <AppRoutes />
    </main>
  </div>
)
export default App
