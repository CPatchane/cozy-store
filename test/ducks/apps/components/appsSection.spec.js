'use strict'

/* eslint-env jest */

import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import { tMock } from '../../../jestLib/I18n'
import { AppsSection } from 'ducks/apps/components/AppsSection'
import SmallAppItem from 'ducks/components/SmallAppItem'

import mockApps from '../_mockApps'

Enzyme.configure({ adapter: new Adapter() })

describe('AppsSection component', () => {
  it('should be rendered correctly with apps list, subtitle and onAppClick', () => {
    const mockOnAppClick = jest.fn()
    const component = shallow(
      <AppsSection
        t={tMock}
        lang='en'
        subtitle='Test Apps'
        appsList={mockApps}
        onAppClick={mockOnAppClick}
      />
    ).getElement()
    expect(component).toMatchSnapshot()
  })

  it('should run provided onAppClick on SmallAppItem click event', () => {
    const mockOnAppClick = jest.fn()
    const component = shallow(
      <AppsSection
        t={tMock}
        lang='en'
        subtitle='Test Apps'
        appsList={mockApps}
        onAppClick={mockOnAppClick}
      />
    )
    expect(component.find(SmallAppItem).length).toBe(mockApps.length)
    const appItem = component
      .find(SmallAppItem)
      .at(0)
      .dive() // shallow on more level on first app item
    appItem.simulate('click')
    expect(mockOnAppClick.mock.calls.length).toBe(1)
    expect(mockOnAppClick.mock.calls[0][0]).toBe(mockApps[0].slug)
  })
})
