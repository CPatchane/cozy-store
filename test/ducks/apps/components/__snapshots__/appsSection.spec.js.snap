// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppsSection component should be rendered correctly with apps list, subtitle and onAppClick 1`] = `
<div
  className="sto-sections-apps"
>
  <h3>
    Test Apps
  </h3>
  <div
    className="sto-sections-list"
  >
    <Wrapper
      app={
        Object {
          "categories": Array [
            "cozy",
          ],
          "developer": Object {
            "name": "Cozy",
          },
          "editor": "Cozy",
          "icon": "<svg></svg>",
          "installed": true,
          "isInRegistry": true,
          "name": "Collect",
          "name_prefix": "Cozy",
          "permissions": Object {
            "mock": Object {
              "type": "io.mock.doctype",
            },
            "mock2": Object {
              "type": "io.mock.doctype2",
            },
          },
          "related": "http://collect.cozy.mock/",
          "slug": "collect",
          "tags": Array [
            "konnector",
            "collect",
            "bills",
            "providers",
            "files",
          ],
          "type": "webapp",
          "uninstallable": false,
          "version": "3.0.0",
          "versions": Object {
            "beta": Array [
              "3.0.0",
            ],
            "dev": Array [
              "3.0.0",
            ],
            "stable": Array [
              "3.0.0",
            ],
          },
        }
      }
      name="Collect"
      namePrefix="Cozy"
      onClick={[Function]}
    />
    <Wrapper
      app={
        Object {
          "categories": Array [
            "others",
          ],
          "developer": Object {
            "name": "Cozy",
          },
          "editor": "Cozy",
          "icon": "<svg></svg>",
          "isInRegistry": true,
          "locales": Object {
            "name": "For Dev Only",
          },
          "name": "DevOnly",
          "slug": "devonly",
          "type": "webapp",
          "uninstallable": true,
          "versions": Object {
            "beta": Array [],
            "dev": Array [
              "1.0.0-betaojdkehy989ekhflldh",
            ],
            "stable": Array [],
          },
        }
      }
      name="DevOnly"
      namePrefix=""
      onClick={[Function]}
    />
    <Wrapper
      app={
        Object {
          "categories": Array [
            "cozy",
          ],
          "developer": Object {
            "name": "Cozy",
          },
          "editor": "Cozy",
          "icon": "<svg></svg>",
          "installed": true,
          "name": "Drive",
          "name_prefix": "Cozy",
          "permissions": Object {
            "mock2": Object {
              "type": "io.mock.doctype2",
            },
          },
          "related": "http://drive.cozy.mock/",
          "slug": "drive",
          "tags": Array [
            "search",
            "files",
            "folders",
          ],
          "type": "webapp",
          "uninstallable": false,
          "version": "3.0.0-beta89bnhj3993",
        }
      }
      name="Drive"
      namePrefix="Cozy"
      onClick={[Function]}
    />
    <Wrapper
      app={
        Object {
          "categories": Array [
            "isp",
            "telecom",
          ],
          "developer": Object {
            "name": "Cozy",
          },
          "icon": "<svg></svg>",
          "isInRegistry": true,
          "name": "Bouilligue",
          "slug": "konnector-bouilligue",
          "type": "konnector",
          "uninstallable": true,
          "versions": Object {
            "beta": Array [
              "0.1.0",
            ],
            "dev": Array [
              "0.1.0",
            ],
            "stable": Array [
              "0.1.0",
            ],
          },
        }
      }
      name="Bouilligue"
      namePrefix=""
      onClick={[Function]}
    />
    <Wrapper
      app={
        Object {
          "categories": Array [
            "transport",
          ],
          "developer": Object {
            "name": "Cozy",
          },
          "icon": "<svg></svg>",
          "installed": true,
          "isInRegistry": true,
          "name": "Trinlane",
          "permissions": Object {
            "mock": Object {
              "type": "io.mock.doctype",
            },
          },
          "slug": "konnector-trinlane",
          "tags": Array [
            "transport",
            "files",
            "bills",
          ],
          "type": "konnector",
          "uninstallable": true,
          "version": "0.1.0",
          "versions": Object {
            "beta": Array [
              "0.1.0",
            ],
            "dev": Array [
              "0.1.0",
            ],
            "stable": Array [
              "0.1.0",
            ],
          },
        }
      }
      name="Trinlane"
      namePrefix=""
      onClick={[Function]}
    />
    <Wrapper
      app={
        Object {
          "categories": Array [
            "cozy",
          ],
          "developer": Object {
            "name": "Cozy",
          },
          "editor": "Cozy",
          "icon": "<svg></svg>",
          "isInRegistry": true,
          "name": "Photos",
          "name_prefix": "Cozy",
          "slug": "photos",
          "type": "webapp",
          "uninstallable": true,
          "versions": Object {
            "beta": Array [
              "3.0.0",
            ],
            "dev": Array [
              "3.0.0",
            ],
            "stable": Array [
              "3.0.0",
            ],
          },
        }
      }
      name="Photos"
      namePrefix="Cozy"
      onClick={[Function]}
    />
    <Wrapper
      app={
        Object {
          "availableVersion": true,
          "categories": Array [
            "partners",
          ],
          "icon": "<svg></svg>",
          "installed": true,
          "isInRegistry": true,
          "name": "Tasky",
          "slug": "tasky",
          "type": "webapp",
          "uninstallable": true,
          "versions": Object {
            "beta": Array [
              "1.0.0",
            ],
            "dev": Array [
              "1.0.0",
            ],
            "stable": Array [
              "1.0.0",
            ],
          },
        }
      }
      name="Tasky"
      namePrefix=""
      onClick={[Function]}
    />
  </div>
</div>
`;
